src/electionguard-ui/ElectionGuard.UI/ViewModels/SettingsViewModel.cs

```diff
@@ -14,13 +16,25 @@ public SettingsViewModel(IServiceProvider serviceProvider) : base(null, serviceP
     [ObservableProperty]
     private string databasePassword = Preferences.Get("DbPassword", "");
 
+    [ObservableProperty]
+    private string databaseConnectionString = Preferences.Get("DbConnection", "");
+
     [RelayCommand]
     private void Save()
     {
         // save db settings and reset the db connection
         Preferences.Set("DbAddress", DatabaseAddress);
         Preferences.Set("DbPassword", DatabasePassword);
-        DbService.Init(DatabaseAddress, DatabasePassword);
+        Preferences.Set("DbConnection", DatabaseConnectionString);
+
+        if (!string.IsNullOrEmpty(DatabaseConnectionString))
+        {
+            DbService.Init(DatabaseConnectionString);

```

Comment:
To reduce user confusion, if we save with a connection string, can we blank out Address / Pw?